reference/src/main/java/com/google/android/fhir/reference/PatientListViewModel.kt

```diff
@@ -70,9 +70,38 @@ class PatientListViewModel(application: Application, private val fhirEngine: Fhi
       .take(MAX_RESOURCE_COUNT)
       .mapIndexed { index, fhirPatient -> fhirPatient.toPatientItem(index + 1) }
       .let { patients.addAll(it) }
+
+    val risks = getRiskAssessments()
+    patients.map { patient ->
+      risks.filter { it?.subject?.reference == "Patient/${patient.resourceId}" }.map {
+        patient.risk = it?.prediction?.first()?.qualitativeRisk?.coding?.first()?.code
+      }
+    }
```

Comment:
Since we would want to show the status based on the latest RiskAssessment of the Patient, below code might be better suited.
```suggestion
      patients.forEach {patient->
      risks.filter { it?.subject?.reference == "Patient/${patient.resourceId}" }.firstOrNull()?.let {
        patient.risk = it?.prediction?.first()?.qualitativeRisk?.coding?.first()?.code
      }
    }
```

Reply:
On the other hand, I feel `getRiskAssessments()` should return a `Map` instead of the `Collection`. Then, the above `risks.filter` can be replaced with a simple `Map.get`.